crates/viewer/re_ui/tests/list_item_tests.rs

```diff
@@ -0,0 +1,205 @@
+                            .value_bool_mut(&mut boolean),
+                    );
+
+                    ui.list_item().show_hierarchical(
+                        ui,
+                        list_item::PropertyContent::new("Text").value_text(&text),
+                    );
+
+                    ui.list_item().show_hierarchical(
+                        ui,
+                        list_item::PropertyContent::new("Text (editable)")
+                            .value_text_mut(&mut text),
+                    );
+
+                    ui.list_item().show_hierarchical(
+                        ui,
+                        list_item::PropertyContent::new("Color")
+                            .with_icon(&re_ui::icons::SPACE_VIEW_TEXT)
+                            .action_button(&re_ui::icons::ADD, || {
+                                re_log::warn!("Add button clicked");
+                            })
+                            .value_color(&color),
+                    );
+
+                    ui.list_item().show_hierarchical(
+                        ui,
+                        list_item::PropertyContent::new("Color (editable)")
+                            .with_icon(&re_ui::icons::SPACE_VIEW_TEXT)
+                            .action_button(&re_ui::icons::ADD, || {
+                                re_log::warn!("Add button clicked");
+                            })
+                            .value_color_mut(&mut color),
+                    );
+                });
+            },
+        );
+
+        ui.list_item().show_hierarchical_with_children(
+            ui,
+            ui.make_persistent_id("other features"),
+            true,
+            list_item::LabelContent::new("Other contents:"),
+            |ui| {
+                ui.list_item().show_hierarchical(
+                    ui,
+                    list_item::DebugContent::default()
+                        .label("DebugContent just shows the content area"),
+                );
+
+                ui.list_item().show_hierarchical(
+                    ui,
+                    list_item::CustomContent::new(|ui, context| {
+                        ui.ctx().debug_painter().debug_rect(
+                            context.rect,
+                            egui::Color32::LIGHT_RED,
+                            "CustomContent delegates to a closure",
+                        );
+                    }),
+                )
+            },
+        );
+    };
+
+    let mut harness = egui_kittest::Harness::builder()
+        .with_size(Vec2::new(700.0, 700.0))
+        .build(|ctx| {
+            egui::SidePanel::right("right_panel").show(ctx, |ui| {
+                ui.set_width(650.0);
+                ui.set_max_width(650.0);
+                re_ui::apply_style_and_install_loaders(ui.ctx());
+                list_item::list_item_scope(ui, "list_item_scope", &mut test_code);
+            });
+        });
```

Comment:
Food for thought: it would be nice if we could set things up so that some screenshots tests (like this one) would also be actually runnable (for additional manual testing).

I don't know how (maybe @lucasmerlin has a good idea?), but it would be cool.
In this case I guess the test could actually call some piece of code shared with `re_ui_example`.

Anyways, not necessary for this PR obviously, just an interesting idea imho

Reply:
Gave this a try, I don't think this will be easily possible 🥲

```
on macOS, `EventLoop` must be created on the main thread!
```
But maybe we could use https://github.com/emilk/eterm? 😄 Perfect usecase for that